Proposal to remove --app-pid and --app-pipe-prefix for out of band commands.

jtrejoespinoza-grid opened the pull request:
#### Testing

The proposal is to remove the parameters `--app-pipe-prefix` and `--app-pid` for CI executions for tests that requiere send out of band commands and only use `--app-pipe` with the full path to the fifo file.

app-pipe is a string which is defined is the linux app. For example to all_clusters:  /tmp/all_clusters_app_fifo_<id> where <id> is  defined environment variable like <ALL_CLUSTERS_APP_ID>. This way we can run locally by just setting a environment variable with the id. For CI we set up static values for ids so they are available during all the execution.

Resume of actions performed.

- Added fail if the app_pipe fifo is not found from start.
- Removed the app_pid and app-pipe-prefix and following dependencies.
- Refactored test cases with manual implementations.
- Updated --app-pipe CI strings for test cases.



## Fixes: https://github.com/project-chip/matter-test-scripts/issues/538

jtrejoespinoza-grid committed:
Updated linux apps to read the FIFO without using the app_pid. Now uses a value from a env variable or static value. This will be usefull on tests to use only -app_pipe <app_prefix>+ instead of two paramters.

jtrejoespinoza-grid committed:
Update write_to_app_pipe to only use app_pipe argument which is the full path to pipe file <prefix>_<id>, raises FileNotFound error if the pipe file is not found from start. Removed arguments app_pid and app_pipe_prefix as they are not needed anymore.

jtrejoespinoza-grid committed:
Removed arguments --app-pid and --app-pipe-prefix

jtrejoespinoza-grid committed:
Updated app-pipe_prefix to app-pipe with APP_ID in test cases

jtrejoespinoza-grid committed:
Updated OPSTATE testcases and OVENOPSTATE test cases to use --app-pippe with id instead of PID of the app.

jtrejoespinoza-grid committed:
Fix not found when the parameter is not available

jtrejoespinoza-grid committed:
Removed unused ci actions

jtrejoespinoza-grid committed:
Added APP_IDs for workflows. Currently those are static values for CI.

jtrejoespinoza-grid committed:
Remove automatically appended --app-pid on execution. Now it dependes on the CI parameters provided by the user.

jtrejoespinoza-grid committed:
Fix typo and restyle

restyled-commits committed:
Restyled by whitespace

restyled-commits committed:
Restyled by clang-format

jtrejoespinoza-grid committed:
fix: Solved typo for TC_RVCRUM_2_1. Added quotes for APP_ID.

jtrejoespinoza-grid committed:
Merge branch 'master' into write_to_app_static_id

jtrejoespinoza-grid committed:
Merge branch 'master' into write_to_app_static_id

jtrejoespinoza-grid committed:
fix: Fixed id for bridge app. Added --app-pipe for BRBINFO. Added missing app-pipe for TC_SWITCH

jtrejoespinoza-grid committed:
Merge branch 'master' into write_to_app_static_id

jtrejoespinoza-grid committed:
Typo in variable name

jtrejoespinoza-grid committed:
Typo RCV_CHIP

jtrejoespinoza-grid committed:
Missing file exists check  when the app_pipe is send from the test case.

gabc-cpp commented on examples/air-quality-sensor-app/linux/main.cpp:
```diff
@@ -52,7 +52,14 @@ int main(int argc, char * argv[])
+        app_id = "54321";
+    }
+
+    std::string app_id_str = app_id;
+    std::string path       = kChipEventFifoPathPrefix + app_id_str;
```
```suggestion
    std::string path = kChipEventFifoPathPrefix + std::string(app_id);
```

This option is leaner, and it might give the compiler an opportunity to optimize the object code.

Try this on the other `.cpp` files as well.

jtrejoespinoza-grid committed:
fix: code review fix, optimize for string.

andy31415 commented on src/python_testing/TC_OCC_3_1.py:
```diff
@@ -27,7 +27,7 @@
 #       --trace-to perfetto:${TRACE_TEST_PERFETTO}.perfetto
 #       --endpoint 1
-#       --app-pipe_prefix /tmp/chip_all_clusters_fifo_
+#       --app-pipe /tmp/chip_all_clusters_fifo_${ALL_CLUSTERS_APP_ID}
```
if we modify these anyway, how about passing both app and script arguments instead?

jtrejoespinoza-grid replied:
Left with only --app-pipe in the CI arguments. No suffix ID added.

andy31415 commented on .github/workflows/tests.yaml:
```diff
@@ -539,15 +539,18 @@ jobs:
                   echo "ALL_CLUSTERS_APP: out/linux-x64-all-clusters-${BUILD_VARIANT}-tsan-clang-test/chip-all-clusters-app" >> /tmp/test_env.yaml
-                  echo "BRIDGE_APP: out/linux-x64-bridge-${BUILD_VARIANT}-tsan-clang-test/chip-bridge-app" >> /tmp/test_env.yaml
-                  echo "CHIP_LOCK_APP: out/linux-x64-lock-${BUILD_VARIANT}-tsan-clang-test/chip-lock-app" >> /tmp/test_env.yaml
+                  echo "ALL_CLUSTERS_APP_ID: \"123452\"" >> /tmp/test_env.yaml
```
I would rather not keep growing this significantly - having a unique ID seems a concern for specific tests - those already specify a app-prefix. Lets place these things into arguments on those scripts rather than global.

The alternative seems to be that for every app we set up a new constant in the list, and it is unclear why we even try to make these unique.

jtrejoespinoza-grid replied:
Removed from yaml file.

andy31415 commented on examples/air-quality-sensor-app/linux/main.cpp:
```diff
@@ -52,7 +52,13 @@ int main(int argc, char * argv[])
     VerifyOrDie(ChipLinuxAppInit(argc, argv) == 0);
 
-    std::string path = kChipEventFifoPathPrefix + std::to_string(getpid());
+    const char * app_id = std::getenv("CHIP_AIR_QUALITY_APP_ID");
```
can we have these a command-line argument? Environment variables are not as obvious.

jtrejoespinoza-grid replied:
Added  argument --app-id for linux apps.

jtrejoespinoza-grid committed:
Removed vars from yaml file

jtrejoespinoza-grid committed:
Removed APP_ID from Test case CI arguments from --app-pipe

jtrejoespinoza-grid committed:
Removed env variable for custom app-id. Replaced for app argument --app-id

restyled-commits committed:
Restyled by whitespace

jtrejoespinoza-grid committed:
Restore change.

jtrejoespinoza-grid committed:
Merge branch 'master' into write_to_app_static_id

jtrejoespinoza-grid committed:
Merge branch 'master' into write_to_app_static_id

ratgr commented on examples/platform/linux/Options.cpp:
```diff
@@ -635,6 +640,10 @@ bool HandleOption(const char * aProgram, OptionSet * aOptions, int aIdentifier,
         break;
 
+    case kDeviceOption_AppID:
+        LinuxDeviceOptions::GetInstance().app_id = aValue;
```
Not Blocking but, This is a little bit dangerous, depending on where aValue comes from it might not exist in the future, would recommend making app_id a std::string, however I see that the pattern is used and used with the arguments.

This is just a little timebomb, eventually someone might call this function (maybe from a unit-test) with a not so safe code, and code will just not work.

khodya replied:
should we resolve this then?

jtrejoespinoza-grid committed:
fix: Fixed from comments, app_id set to empty string instead of nullptr, this will avoid uneeded check on each of the apps.

jtrejoespinoza-grid commented on examples/platform/linux/Options.h:
```diff
@@ -74,6 +74,7 @@ struct LinuxDeviceOptions
     const char * command                = nullptr;
     const char * PICS                   = nullptr;
     const char * KVS                    = nullptr;
+    const char * app_id                 = nullptr;
```
Updated to empty string.

jtrejoespinoza-grid committed:
fix: removed comma

juandediosg reviewed (approved):
Fixed: the comma has been removed.

andy31415 commented on examples/platform/linux/Options.h:
```diff
@@ -74,6 +74,7 @@ struct LinuxDeviceOptions
     const char * command                = nullptr;
     const char * PICS                   = nullptr;
     const char * KVS                    = nullptr;
+    const char * app_id                 = "";
```
We still seem to have hard-coded paths for each example applications. Let's do a change where we "pass in the path to the pipes". Applications can have defaults if needed and those defaults could be even be PID based for, however our tests should explicitly pass in the FIFO path to BOTH applications and test scripts.

andy31415 reviewed (changes requested):
Please build the fifo in the following way:

- [x] test runner should be passed in the FIFO (you already do that, but have an odd "_" because of the way applications concatenate an "application custom id" ... fix that)
- [x] do NOT use concatenation anymore. have example applications do:
   - [ ] have a default fifo path. Can stay as whatever (even missing if we want and not start the fifo). I would maybe even prefer no fifo by default if not asked for it
   - [ ] have the application receive a "fifo path"/"pipe path" as input argument. This is important. Do not do "prefix" or "custom id" or things like that. Get the full path as an argument and use that
- [x] update tests to pass in the same fifo to BOTH test and app. I would suggest naming it based on test case (like `/tmp/bininfo_3_2_fifo` and stop assuming hardcoded paths inside example applications)

jtrejoespinoza-grid committed:
Added app-pipe from scripts args to app-args if present. This is to avoid having the fifo path in scripts args and app-args and avoid posible run errors.

jtrejoespinoza-grid committed:
Added custom fifo path for each one of the test cases.

jtrejoespinoza-grid committed:
Removed prefix. Updated app-id to app-pipe. Now app pipe gets the full path of the fifo path. If not defined avoid sChipNamedPipeCommands.Start()

jtrejoespinoza-grid committed:
Merge branch 'master' into write_to_app_static_id

restyled-commits committed:
Restyled by clang-format

jtrejoespinoza-grid committed:
Updated empty string compare

jtrejoespinoza-grid commented:
Updated the following.

Updated linux apps: updated `--app-id` to `--app-pipe` now there is no prefix or appending is just the path for the fifo. If not defined does not start the sChipNamedPipeCommands.Start()

Updated --app-pipe to custom pipe name for each of the test cases that required --app-pipe. Removed previous default pipe for for each one of the linux apps.

Updated run_python_test.py to use the --app-pipe argument and if present and add it into the app-args this is in order to make it work for the CI jobs.

jtrejoespinoza-grid committed:
Added --app-pipe into app-args. Removed auto append --app-pipe this will need repeated value but will not force same fifo and allow more scenarios.

jtrejoespinoza-grid committed:
Merge branch 'master' into write_to_app_static_id

jtrejoespinoza-grid committed:
Updated tc_soil_2_2 with new fifo format.
